launchflow/aws/ecs_fargate.py

```diff
@@ -158,46 +172,203 @@ def __init__(
         self.memory = memory
         self.port = port
         self.desired_count = desired_count
+        self.dockerfile = dockerfile
 
-    def inputs(self) -> ECSFargateInputs:
-        return ECSFargateInputs(
+    def inputs(self) -> ECSFargateServiceInputs:
+        return ECSFargateServiceInputs(
             cpu=self.cpu,
             memory=self.memory,
+            port=self.port,
         )
 
     def resources(self) -> List[Resource]:
-        to_return = [
+        to_return: List[Resource] = [
             self._ecr,
             self._code_build_project,
-            self._ecs_cluster,
             self._ecs_fargate_service_container,
             self._alb,
         ]
-        if self._https_certificate:
-            to_return.append(self._https_certificate)
+        if self._manage_ecs_cluster:
+            to_return.append(self._ecs_cluster)
         return to_return  # type: ignore
 
-    def outputs(self) -> AWSDockerServiceOutputs:
+    def outputs(self) -> ServiceOutputs:
         try:
-            ecr_outputs = self._ecr.outputs()
-            code_build_outputs = self._code_build_project.outputs()
             fargate_outputs = self._ecs_fargate_service_container.outputs()
             alb_outputs = self._alb.outputs()
         except exceptions.ResourceOutputsNotFound:
             raise exceptions.ServiceOutputsNotFound(service_name=self.name)
 
         service_url = f"http://{alb_outputs.alb_dns_name}"
-        if self._https_certificate:
-            domain = self._https_certificate.outputs().domain_name
-            service_url = f"https://{domain}"
 
-        service_outputs = AWSDockerServiceOutputs(
+        service_outputs = ServiceOutputs(
             service_url=service_url,
-            docker_repository=ecr_outputs.repository_url,
-            code_build_project_name=code_build_outputs.project_name,
             # TODO: Support custom domains for ECS Fargate
             dns_outputs=None,
         )
         service_outputs.aws_arn = fargate_outputs.aws_arn
 
         return service_outputs
+
+    async def _build(
+        self,
+        *,
+        aws_environment_config: AWSEnvironmentConfig,
+        launchflow_uri: LaunchFlowURI,
+        deployment_id: str,
+        build_log_file: IO,  # TODO: Update this service to use the build_log_file isntead of creating a new one
```

Comment:
Can you file issues for these todos so we don't lose track of them?

Reply:
updated ecs to use the log file in this pr, going to do the others this week with the gcp service refactor